scripts/masker/dma-collection-masker

```diff
@@ -208,24 +329,40 @@ def _generate_identity_map_from_rule(
     if existing_identity_map is None:
         existing_identity_map = {}
 
-    all_values: List[str] = list(existing_identity_map.keys())
+    all_values: list[str] = list(existing_identity_map.keys())
     with collection_file.open(mode="r", encoding="utf-8") as f:
         data_to_mask = csv.reader(f, delimiter="|", quotechar='"')
         for rn, row in enumerate(data_to_mask):
             if rn > 0:
                 # Get the value from column to mask
                 all_values.append(row[rule.column_position])
     unique_values = list(dict.fromkeys(all_values))
+    match rule.mask_type:
```

Comment:
I think we need to at least support all current stable releases, including python 3.8 and 3.9.  I believe match case was introduced in 3.10.  Let's see if there's a backport that can be used for <3.10.

Reply:
There was no need to use the `match case` specifically so I've removed it with an implementation that adds the fake data generator function as an attribute to the rule, which I think is nicer.